lib/steep/type_construction.rb

```diff
@@ -3465,6 +3466,24 @@ def type_send(node, send_node:, block_params:, block_body:, unwrap: false, tapp:
         end
       end
 
+      if module_context && receiver.nil?
+        case method_name
+        when :attr_reader
+          arguments.each do |arg|
+            module_context.defined_instance_methods << arg.children[0]
```

Comment:
Looks like it's assuming the arguments are symbol literals? Should consider the case they are not.